File: src/test/java/com/mbayou/kick4k/events/ModerationBannedEventTest.java
package com.mbayou.kick4k.events;

import org.junit.jupiter.api.Test;
import com.mbayou.kick4k.events.type.BanMetadata;
import com.mbayou.kick4k.events.type.EventUser;
import com.mbayou.kick4k.events.type.ModerationBannedEvent;

import java.time.Instant;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;

public class ModerationBannedEventTest extends KickEventDispatcherTest {
    private static final String PAYLOAD = """
            {
              "broadcaster": {
                "is_anonymous": false,
                "user_id": 123456789,
                "username": "broadcaster_name",
                "is_verified": true,
                "profile_picture": "https://example.com/broadcaster_avatar.jpg",
                "channel_slug": "broadcaster_channel",
                "identity": null
              },
              "moderator": {
                "is_anonymous": false,
                "user_id": 987654321,
                "username": "moderator_name",
                "is_verified": false,
                "profile_picture": "https://example.com/moderator_avatar.jpg",
                "channel_slug": "moderator_channel",
                "identity": null
              },
              "banned_user": {
                "is_anonymous": false,
                "user_id": 135790135,
                "username": "banned_user_name",
                "is_verified": false,
                "profile_picture": "https://example.com/banned_user_avatar.jpg",
                "channel_slug": "banned_user_channel",
                "identity": null
              },
              "metadata": {
                "reason": "banned reason",
                "created_at": "2025-01-14T16:08:05Z",
                "expires_at": "2025-01-14T16:10:06Z"
              }
            }""";

    @Test
    public void should_deserialize_moderation_banned_event() {
        ModerationBannedEvent event = dispatchAndCapture(
                ModerationBannedEvent.class,
                "moderation.banned",
                "1",
                PAYLOAD
        );

        EventUser broadcaster = event.getBroadcaster();
        assertNotNull(broadcaster);
        assertFalse(broadcaster.getAnonymous());
        assertEquals(123456789, broadcaster.getUserId());
        assertEquals("broadcaster_name", broadcaster.getUsername());

        EventUser moderator = event.getModerator();
        assertNotNull(moderator);
        assertEquals(987654321, moderator.getUserId());
        assertEquals("moderator_name", moderator.getUsername());

        EventUser bannedUser = event.getBannedUser();
        assertNotNull(bannedUser);
        assertEquals(135790135, bannedUser.getUserId());
        assertEquals("banned_user_name", bannedUser.getUsername());

        BanMetadata metadata = event.getMetadata();
        assertNotNull(metadata);
        assertEquals("banned reason", metadata.getReason());
        assertEquals(Instant.parse("2025-01-14T16:08:05Z"), metadata.getCreatedAt());
        assertEquals(Instant.parse("2025-01-14T16:10:06Z"), metadata.getExpiresAt());
    }
}

File: src/test/kotlin/com/mbayou/kick4k/events/ModerationBannedEventTest.kt
package com.mbayou.kick4k.events

import com.mbayou.kick4k.events.type.ModerationBannedEvent
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import java.time.Instant

class ModerationBannedEventTest : KickEventDispatcherTest() {
    @Test
    fun shouldDeserializeModerationBannedEvent() {
        val event = dispatchAndCapture(
            ModerationBannedEvent::class.java,
            "moderation.banned",
            "1",
            PAYLOAD,
        )

        val broadcaster = event.broadcaster
        assertNotNull(broadcaster)
        assertFalse(broadcaster!!.isAnonymous!!)
        assertEquals(123456789, broadcaster.userId)
        assertEquals("broadcaster_name", broadcaster.username)

        val moderator = event.moderator
        assertNotNull(moderator)
        assertEquals(987654321, moderator!!.userId)
        assertEquals("moderator_name", moderator.username)

        val bannedUser = event.bannedUser
        assertNotNull(bannedUser)
        assertEquals(135790135, bannedUser!!.userId)
        assertEquals("banned_user_name", bannedUser.username)

        val metadata = event.metadata
        assertNotNull(metadata)
        assertEquals("banned reason", metadata!!.reason)
        assertEquals(Instant.parse("2025-01-14T16:08:05Z"), metadata.createdAt)
        assertEquals(Instant.parse("2025-01-14T16:10:06Z"), metadata.expiresAt)
    }

    companion object {
        private const val PAYLOAD = """
            {
              "broadcaster": {
                "is_anonymous": false,
                "user_id": 123456789,
                "username": "broadcaster_name",
                "is_verified": true,
                "profile_picture": "https://example.com/broadcaster_avatar.jpg",
                "channel_slug": "broadcaster_channel",
                "identity": null
              },
              "moderator": {
                "is_anonymous": false,
                "user_id": 987654321,
                "username": "moderator_name",
                "is_verified": false,
                "profile_picture": "https://example.com/moderator_avatar.jpg",
                "channel_slug": "moderator_channel",
                "identity": null
              },
              "banned_user": {
                "is_anonymous": false,
                "user_id": 135790135,
                "username": "banned_user_name",
                "is_verified": false,
                "profile_picture": "https://example.com/banned_user_avatar.jpg",
                "channel_slug": "banned_user_channel",
                "identity": null
              },
              "metadata": {
                "reason": "banned reason",
                "created_at": "2025-01-14T16:08:05Z",
                "expires_at": "2025-01-14T16:10:06Z"
              }
            }
        """
    }
}
